eventbus/jetstream/base/jetstream.go

```diff
@@ -0,0 +1,133 @@
+package base
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/argoproj/argo-events/common"
+	eventbuscommon "github.com/argoproj/argo-events/eventbus/common"
+	eventbusv1alpha1 "github.com/argoproj/argo-events/pkg/apis/eventbus/v1alpha1"
+	nats "github.com/nats-io/nats.go"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+	"go.uber.org/zap"
+)
+
+type Jetstream struct {
+	url  string
+	auth *eventbuscommon.Auth
+
+	MgmtConnection JetstreamConnection
+
+	streamSettings string
+
+	Logger *zap.SugaredLogger
```

Comment:
Same. It's being used by SourceJetstream and SensorJetstream.

Reply:
don't worry too much about prior-art, if a variable should be private, make it private

Reply:
So, the reason it's public is because the SourceJetstream and SensorJetstream structs have embedded the Jetstream struct and are accessing those members directly. 

Now, if all 3 structs were in the same package, I assume this would not be an issue and it could be lower case. Right now they are in separate directories each with a package name that corresponds to the directory, but I guess that doesn't need to be the case?

Let me know how you'd like me to progress.